swim/join_sender.go

```diff
@@ -388,7 +388,11 @@ func (j *joinSender) JoinGroup(nodesJoined []string) ([]string, []string) {
 					failed = true
 					break
 				}
+
 				j.node.memberlist.Update(res.Membership)
+				for _, change := range res.Membership {
```

Comment:
Yeah I've wondered about the same thing. I figured this is the most obvious way to do it since it's reusing the elements that are already built. Will add comment though